ouroboros-consensus-mock/src/Ouroboros/Consensus/Mock/Protocol/Praos.hs

```diff
@@ -78,31 +78,97 @@ import           Ouroboros.Consensus.Protocol.Abstract
 import           Ouroboros.Consensus.Protocol.Signed
 import           Ouroboros.Consensus.Util.Condense
 
+-- The Praos paper can be located at https://ia.cr/2017/573
+--
+-- A block as defined in Praos (section 3, definition 2, then extended in Fig 9)
+-- consist of a tuple:
+--
+-- > B = (slⱼ, st, d, B_πj, ρ, σⱼ)
+--
+-- where
+-- - slⱼ: the slot at which the block was generated.
+--
+-- - st: the current state.
+--
+-- - d: data (transaction data in most cases).
+--
+-- - B_πj: block proof consisting of (Uᵢ, y, π).
+--      - y: a VRF output used to confirm tha Uᵢ was the slot leader.
+--      - π: the VRF proof of the above value.
+--
+--      > (y, π) ≜ VRF_evalProve(η, slⱼ, TEST) see Fig 9
+--
+-- - ρ: the block nonce consisting of (ρ_y, ρ_π), to capture entropy from the
+--      block forging process.
+--      - ρ_y: a VRF output used to confirm this block captured all the previous
```

Comment:
I'm feeling clueless about rho_y; this is a part of the crypto details I've not attended to. Let's chat about it on a call.

I think this comment is useful, even without fully grokking it.

P.S. - Aha! I see (from your great comment below) in Fig. 10 that these rhos are ultimately used to build-up the eta. I'm familiar with eta, so now this suddenly makes a lot more sense to me.

Reply:
I guess from this comment that it was okey then? :smile: